class.phpmailer.php

```diff
@@ -3274,15 +3274,20 @@ public function fixEOL($str)
      * @access public
      * @param string $name Custom header name
      * @param string $value Header value
+     * @param bool $overwrite Whether to overwrite if a header with name $name exists.
      * @return void
      */
-    public function addCustomHeader($name, $value = null)
+    public function addCustomHeader($name, $value = null, $overwrite = false)
     {
         if ($value === null) {
-            // Value passed in as name:value
-            $this->CustomHeader[] = explode(':', $name, 2);
-        } else {
-            $this->CustomHeader[] = array($name, $value);
+            $header = explode(':', $name, 2);
+            $name = $header[0];
+        }
+        if ($overwrite) {
+            $this->CustomHeader[$name] = isset($header) ? $header : array($name, $value);
+        }
+        else {
+            $this->CustomHeader[] = isset($header) ? $header : array($name, $value);
         }
```

Comment:
This logic doesn't work. If you've already added a header (without overwrite) then it won't get a named index. if you call it again with overwrite, you'll get a second copy because there is no matching index. There's also a problem that you can't have an assoc array with duplicate keys. You will need to iterate through the array and find a matching name.